### PR TITLE
Fix a % format mismatch that clang 10 was complaining about for comm=ofi

### PR DESCRIPTION
**Pull request by bradcray**
Trying CHPL_COMM=ofi on my Mac with CHPL_DEVELOPER set, I was getting
a broken build due to a % format mismatch.  This resolves it by using the stdint.h
macros for fixed-size int formats.

### CODE REVIEW

**Comment by bradcray** [comment]
@gbtitus / @jhh67: This resolves the compilation error I was getting on my Mac for `CHPL_COMM=ofi`.  However, testing on `chapcs11`, I'm getting a host of other compilation errors, e.g.

```
comm-ofi.c: In function ‘setCheckMsgOrderFenceProv’:
comm-ofi.c:1443:30: error: ‘FI_ORDER_ATOMIC_RAW’ undeclared (first use in this function); did you mean ‘FI_TYPE_ATOMIC_OP’?
   uint64_t need_msg_orders = FI_ORDER_ATOMIC_RAW
                              ^~~~~~~~~~~~~~~~~~~
                              FI_TYPE_ATOMIC_OP
```

It's admittedly been awhile since I've built CHPL_COMM=ofi on chapcs11, so it could be a personal problem.

File: runtime/src/comm/ofi/comm-ofi.c [hunk 1]
```diff
@@ -2414,7 +2414,7 @@ chpl_bool nextMemMapEntry(void** pAddr, size_t* pSize,
     int ch;
 
     int scn_cnt;
-    scn_cnt = fscanf(f, "%lx-%lx%4s%*x%*x:%*x%*x", &lo_addr, &hi_addr, perms);
+    scn_cnt = fscanf(f, "%"SCNu64"-%"SCNu64"%4s%*x%*x:%*x%*x", &lo_addr, &hi_addr, perms);
```
Comment (gbtitus) [inline runtime/src/comm/ofi/comm-ofi.c:2417]: This is the right solution, but would you mind putting spaces on both sides of the `SCNu64` tokens? That would match the formatting elsewhere in the file.

**Comment by gbtitus** [comment]
If `FI_ORDER_ATOMIC_RAW` is not defined, I suspect you're using a really old libfabric. What's the `CHPL_LIBFABRIC` setting in your `printchplenv` output? We have to use "bundled" on chapcs11.

**Comment by bradcray** [comment]
@gbtitus : Thanks for reviewing.  This was with `CHPL_LLVM=system` on my Mac using (what I believe is) libfabric 1.8.1.  When I set `CHPL_COMM=ofi`, `CHPL_LLVM=system` was chosen as the default which is how I ended up here.  But I get the same compile error when using `CHPL_LLVM=bundled` (which seems right since I think that this code isn't in an `#ifdef`-protected region, is it?  i.e., the C compiler would see it with any setting of CHPL_LLVM?)

File: runtime/src/comm/ofi/comm-ofi.c (hunk 1 above)
Comment (bradcray) [reply runtime/src/comm/ofi/comm-ofi.c]: Should be done now if you want to double-check it.

**Comment by gbtitus** [comment]
@bradcray My question about your `CHPL_LIBFABRIC` setting was in regard to your report of this error message on chapcs11:

```
comm-ofi.c: In function ‘setCheckMsgOrderFenceProv’:
comm-ofi.c:1443:30: error: ‘FI_ORDER_ATOMIC_RAW’ undeclared (first use in this function); did you mean ‘FI_TYPE_ATOMIC_OP’?
   uint64_t need_msg_orders = FI_ORDER_ATOMIC_RAW
                              ^~~~~~~~~~~~~~~~~~~
                              FI_TYPE_ATOMIC_OP
```

That shouldn't have anything to do with debug or LLVM. It's due to using a too-old `<rdma/fabric.h>`. `CHPL_LIBFABRIC=bundled` should fix it, if you don't already have that set.

**Review by gbtitus — APPROVED** [review]
👍

**Comment by bradcray** [comment]
> My question about your CHPL_LIBFABRIC setting was in regard to your report of this error message on chapcs11:

Ah yes, sorry.  I forgot that I'd posted that comment here rather than on Slack, and didn't connect it to the chapcs11 failure mode rather than the PR's purpose.  You're correct of course that using `CHPL_LIBFABRIC=bundled` solves this.

This raises an interesting question for me:  Why does `CHPL_LIBFABRIC` default to `system` rather than `bundled`?  Checking the docs, it seems surprising that it would given that `system` is barely documented, and in a way that implies it's more developer-oriented than anything?

**Comment by gbtitus** [comment]
> ...  Why does `CHPL_LIBFABRIC` default to `system` rather than `bundled`?  ...

It's a mix of historical + simplicity. When we first defined `CHPL_LIBFABRIC` we didn't even have a bundled libfabric, and that config only had two settings: 'none' and 'system'. Then we added the bundled one, but we left the default 'system' iff `pkg-config` is present and reports the presence of a libfabric install. This gives us the right result on on HPE Cray EX systems with their Slingshot networks, where using the system libfabric is actually required, but it gives the wrong result on CS systems including chapcs, where there is a system-installed libfabric but it's really old. It would be easy to adjust the default to be "bundled" for CS systems, of course.

**Comment by bradcray** [comment]
OK, thanks for the context.  How about if I fire off an issue for this?  (where my intuition would be to take a slightly different approach, but I'll sketch out some possibilities in the issue itself).

**Comment by gbtitus** [comment]
> How about if I fire off an issue for this?

That sounds great, thanks!

And I'm sorry to realize this so late, but could I get you to adjust your commit one more time? The lines of text we're reading from `/proc/self/maps` in the code you're modifying should specifically list the low and high addresses of the memory ranges in hex. In the interest of being picky about what we're reading, and to force us to revisit this code if this area of Linux changes, could you change your `SCNu64` tokens to `SCNx64` instead? Then they'll only match hex values, not any of decimal/octal/hex values as the 'u' ones do.

**Comment by gbtitus** [comment]
After applying a local fix to line 2390 I get a related error on line 2390. Do you see this?

```
comm-ofi.c:2390:24: error: format specifies type 'unsigned long long' but the argument has type 'size_t' (aka 'unsigned long') [-Werror,-Wformat]
                 addr, size, path);
                 ~~~~~~^~~~~~~~~~~
./comm-ofi-internal.h:120:29: note: expanded from macro 'DBG_PRINTF'
      DBG_DO_PRINTF(fmt, ## __VA_ARGS__);                               \
      ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
./comm-ofi-internal.h:112:44: note: expanded from macro 'DBG_DO_PRINTF'
            chpl_comm_ofi_dbg_prefix(), ## __VA_ARGS__);                \
                                           ^~~~~~~~~~~
```

We need to change `" PRIx64 "` to `zx` on 2390.

These are all due to my not building on Mac because (a) EX was the goal and (b) you (I think it was) had reported ofi wasn't working on Mac anyway, so I wasn't going to bother with that platform until I felt I had time to figure out what was wrong.

**Comment by bradcray** [comment]
I forked off the question about CHPL_LIBFABRIC's default to https://github.com/Cray/chapel-private/issues/2115.  I made this a private issue since the setting itself isn't intended to be user-facing.

**Comment by bradcray** [comment]
> After applying a local fix to line 2390 I get a related error on line 2390. Do you see this?

I'm not seeing that.  You may be using a more recent version of clang than I am, though?

I'm not deeply worried about these, but figured I'd do some housekeeping while here and trying to build on Mac.  I can apply the SCNu64->SCNx64 changes, but if it makes more sense for you to take this over and for me to close this PR, that'd be fine too.  Just trying to be a good citizen last week.

**Comment by gbtitus** [comment]
I'd say once the SCNu64 --> SCNx64 change is in, go ahead and merge it and then I'll take over anything needed after that. And thanks again!

**Pull request merged**